sdk/storage/azure-storage-blob/azure/storage/blob/aio/container_client_async.py

```diff
@@ -764,6 +767,119 @@ def walk_blobs(
+        :param lease:
+            Required if the blob has an active lease. Value can be a Lease object
+            or the lease ID as a string.
+        :type lease: ~azure.storage.blob.lease.LeaseClient or str
+        :param str delete_snapshots:
+            Required if the blob has associated snapshots. Values include:
+             - "only": Deletes only the blobs snapshots.
+             - "include": Deletes the blob along with all snapshots.
+        :param datetime if_modified_since:
+            A DateTime value. Azure expects the date value passed in to be UTC.
+            If timezone is included, any non-UTC datetimes will be converted to UTC.
+            If a date is passed in without timezone info, it is assumed to be UTC.
+            Specify this header to perform the operation only
+            if the resource has been modified since the specified time.
+        :param datetime if_unmodified_since:
+            A DateTime value. Azure expects the date value passed in to be UTC.
+            If timezone is included, any non-UTC datetimes will be converted to UTC.
+            If a date is passed in without timezone info, it is assumed to be UTC.
+            Specify this header to perform the operation only if
+            the resource has not been modified since the specified date/time.
+        :param str if_match:
+            An ETag value, or the wildcard character (*). Specify this header to perform
+            the operation only if the resource's ETag matches the value specified.
+        :param str if_none_match:
+            An ETag value, or the wildcard character (*). Specify this header
+            to perform the operation only if the resource's ETag does not match
+            the value specified. Specify the wildcard character (*) to perform
+            the operation only if the resource does not exist, and fail the
+            operation if it does exist.
+        :param int timeout:
+            The timeout parameter is expressed in seconds.
+        :rtype: None
+        """
+        options = BlobClient._generic_delete_blob_options(
+            delete_snapshots=delete_snapshots,
+            lease=lease,
+            timeout=timeout,
+            **kwargs
+        )
+        query_parameters, header_parameters = self._generate_delete_blobs_options(**options)
+
+        reqs = []
+        for blob in blobs:
+            req = HttpRequest(
+                "DELETE",
+                "/{}/{}".format(self.container_name, blob),
+                headers=header_parameters
+            )
+            req.format_parameters(query_parameters)
+            reqs.append(req)
+
+        request = self._client._client.post(
+            url='https://{}/?comp=batch'.format(self.primary_hostname),
+            headers={
+                'x-ms-version': "2019-02-02"
+            }
+        )
+
+        request.set_multipart_mixed(
+            *reqs,
+            policies=[
+                StorageHeadersPolicy(),
```

Comment:
It seems we need to set 'Content-Length' to 0 for every sub-request to make the batch delete request succeed. I feel like currently we are not setting the length of  every sub-request, would you like to double check(I just feel like) or add {'Centent-Legth': '0'} as a parameter of StorageHeadersPolicy constructor?

Reply:
Content-Length should not be added to StorageHeadersPolicy, since the contract of this policy should be limited to Storage specific headers of type "x-ms".

Also, I tested without Content-Length and it works fine :)